src/Raven.Server/Smuggler/Documents/ShardedDatabaseSmuggler.cs

```diff
@@ -0,0 +1,773 @@
+        }
+
+        protected override async Task<SmugglerProgressBase.Counts> ProcessConflictsAsync(SmugglerResult result)
+        {
+            result.Conflicts.Start();
+            try
+            {
+                foreach (var des in _destinations)
+                {
+                    _actionsList.Add(des.Conflicts());
+                }
+                await foreach (var conflict in _source.GetConflictsAsync(_options.Collections))
+                {
+                    HandleImportBatch("Conflicts");
+                    var index = ShardHelper.GetShardIndexforDocument(_transactionOperationContext, _shardAllocation, conflict.Id);
+                    result.Conflicts.LastEtag = conflict.Etag;
+                    await ((IDocumentActions)_actionsList[index]).WriteConflictAsync(conflict, result.Conflicts);
+                    _itemsInBatch++;
+                }
+            }
+            finally
+            {
+                await CleanActionList();
+            }
+
+            return result.Conflicts;
+        }
+
+        protected override async Task<SmugglerProgressBase.Counts> ProcessIndexesAsync(SmugglerResult result)
+        {
+            result.Indexes.Start();
+
+            var configuration = _server.DatabasesLandlord.CreateConfiguration(_databaseRecord);
+
+            await foreach (var index in _source.GetIndexesAsync())
+            {
+                _token.ThrowIfCancellationRequested();
+                result.Indexes.ReadCount++;
+
+                if (index == null)
+                {
+                    result.Indexes.ErroredCount++;
+                    continue;
+                }
+
+                switch (index.Type)
+                {
+                    case IndexType.AutoMap:
+                        await PutAutoMapIndex(result, index, configuration);
+                        break;
+
+                    case IndexType.AutoMapReduce:
+                        await PutAutoMapReduceIndex(result, index, configuration);
+                        break;
+
+                    case IndexType.Map:
+                    case IndexType.MapReduce:
+                    case IndexType.JavaScriptMap:
+                    case IndexType.JavaScriptMapReduce:
+                        await PutIndex(result, index, configuration);
+                        break;
+
+                    case IndexType.Faulty:
+                        break;
+
+                    default:
+                        throw new NotSupportedException(index.Type.ToString());
+                }
+
+                if (result.Indexes.ReadCount % 10 == 0)
+                {
+                    var message = $"Read {result.Indexes.ReadCount:#,#;;0} indexes.";
+                    AddInfoToSmugglerResult(result, message);
+                }
+            }
+            return result.Indexes;
+        }
+
+        private async Task PutAutoMapReduceIndex(SmugglerResult result, IndexDefinitionAndType index, RavenConfiguration configuration)
```

Comment:
Why is that different from the usual behavior of the smuggler? Isn't that already abstracted?

Reply:
At the usual smuggler we use the indexStore and DocumentDatabase

Reply:
Need to match the behavior. Either use the existing one, or modify the normal to have the same thing